### ISSUE TITLE
[docs] Implementors section of Sync (and other similar traits) should separate implementors and !implementors

### ISSUE THREAD

**Issue by alex** [rust-lang/rust#51129]
![image](https://user-images.githubusercontent.com/772/40619532-6af752b4-6263-11e8-8627-2ae03dfe621d.png)

The `!Sync` implementations and the `Sync` implementations really ought to be separated apart, otherwise it makes it harder to skim.

**Comment by QuietMisdreavus** [comment]
Would it work out if the impls were just sorted apart, rather than giving an entirely new section heading to "negative impls"? That might be easier to implement.

**Comment by alex** [comment]
@QuietMisdreavus yes, just sorted seperately would be A+

**Comment by QuietMisdreavus** [comment]
Excellent!

Some mentoring instructions:

The code that displays all these impls is right here:

https://github.com/rust-lang/rust/blob/5205ae8bdc2991eecf3bfbb58ed8f56c0673e738/src/librustdoc/html/render.rs#L2860-L2900

All the collections it's partitioning at the start of that contain `render::Impl` structs, and you can see whether it's a `!impl` by checking the "polarity" of the `clean::Impl` it contains. You can check `implementor.impl_item().polarity` and compare it against `Some(clean::ImplPolarity::Negative)`, and sort these collections before iterating (or partition them farther, heh) to print one before the other. Given the nature of `Send` and `Sync`, i think it would be better to print the negative impls before the positive ones.

**Comment by SLaabsDev** [comment]
I can start on this one

**Comment by QuietMisdreavus** [comment]
@SLaabsDev Thanks! Were the instructions i gave a good start? If you need any help, please ask me!

**Comment by SLaabsDev** [comment]
Yes, the instructions were very helpful. I have it to a point where it is sorting with negative implementors showing first. One question I have is outside of the polarity does the order of implementors need to be preserved?

**Comment by QuietMisdreavus** [comment]
@SLaabsDev That's probably best addressed by all of @rust-lang/docs but what i would say is "don't worry about it", since we currently don't do a very good job of keeping that ordering in the first place. This could be a good place to enforce a good ordering, but you don't have to do that in your PR. (If you asking to choose between `sort` and `sort_unstable`, go ahead and pick `sort` just so we don't disrupt the current ordering *too* much.)

**Comment by hkBst** [comment]
Any update here?

**Comment by GuillaumeGomez** [comment]
Still in progress.

**Comment by GuillaumeGomez** [comment]
Opened [#142380](https://github.com/rust-lang/rust/pull/142380).

**Comment by alex** [comment]
Thanks for looking at this after all these years!

**Comment by GuillaumeGomez** [comment]
If you're patient enough, everything (hopefully) will be done at some point. 😆
